Compilation warning

Getting the following warning when building erlang sources on Linux:

uname -a:  `Linux ... 5.10.89-1-MANJARO #1 SMP PREEMPT`
gcc version: `11.1.0`

```bash
$ ./configure
$ make
...
 CC     obj/x86_64-pc-linux-gnu/opt/jit/erl_os_monotonic_time_extender.o
In file included from sys/common/erl_mmap.c:25:
In function ‘sys_memcpy’,
    inlined from ‘remap_move’ at sys/common/erl_mmap.c:1830:5,
    inlined from ‘erts_mremap’ at sys/common/erl_mmap.c:1900:13:
beam/sys.h:1088:12: warning: argument 2 null where non-null expected [-Wnonnull]
 1088 |     return memcpy(dest,src,n);
      |            ^~~~~~~~~~~~~~~~~~
In file included from sys/unix/erl_unix_sys.h:32,
                 from beam/sys.h:120,
                 from sys/common/erl_mmap.c:25:
sys/common/erl_mmap.c: In function ‘erts_mremap’:
/usr/include/string.h:43:14: note: in a call to function ‘memcpy’ declared ‘nonnull’
   43 | extern void *memcpy (void *__restrict __dest, const void *__restrict __src,
      |              ^~~~~~
 CC     obj/x86_64-pc-linux-gnu/opt/jit/erl_poll.flbk.o
...
```

Looking at lines 1839-1900 of erl_mmap.c, after `void *new_ptr = NULL` is declared and `superaligned` is false, it looks like there is a code path before the call to remap_move on line 1900, in which `new_ptr` is still unassigned.

Also I ran compilation with clang (v13.0.0), and it spotted a few more warnings, which you might want to clean up:
```
beam/erl_process.c:5334:2: warning: variable 'sum__' set but not used [-Wunused-but-set-variable]
        ERTS_DBG_CHK_FULL_REDS_HISTORY(rq);
        ^
beam/erl_process.c:4659:9: note: expanded from macro 'ERTS_DBG_CHK_FULL_REDS_HISTORY'
    int sum__ = 0;                                                      \
        ^
beam/erl_process.c:4785:41: warning: variable 'oowc' set but not used [-Wunused-but-set-variable]
    int forced, active, current_active, oowc, half_full_scheds, full_scheds,
                                        ^
asmjit/core/emitterutils.cpp:42:10: warning: variable 'currentSize' set but not used [-Wunused-but-set-variable]
  size_t currentSize = sb.size();
         ^
beam/erl_db_util.c:4171:9: warning: variable 'heap_used' set but not used [-Wunused-but-set-variable]
    int heap_used = 0;
        ^
```

--- a/erts/emulator/beam/erl_db_util.c
+++ b/erts/emulator/beam/erl_db_util.c
@@ -4168,7 +4168,6 @@ static DMCRet dmc_all_bindings(DMCContext *context,
 			       int *constant)
 {
     int i;
-    int heap_used = 0;
 
     DMC_PUSH(*text, matchPushC);
     DMC_PUSH(*text, NIL);
@@ -4176,7 +4175,6 @@ static DMCRet dmc_all_bindings(DMCContext *context,
 	if (heap->vars[i].is_bound) {
 	    dmc_add_pushv_variant(context, heap, text, i);
 	    DMC_PUSH(*text, matchConsB);
-	    heap_used += 2;
 	}
     }
     ++context->stack_used;

--- a/erts/emulator/beam/erl_process.c
+++ b/erts/emulator/beam/erl_process.c
@@ -4616,7 +4616,6 @@ typedef struct {
     int full_reds;
     int full_reds_history_sum;
     int full_reds_history_change;
-    int oowc;
     int max_len;
 #if ERTS_HAVE_SCHED_UTIL_BALANCING_SUPPORT
     int sched_util;
@@ -4786,7 +4785,7 @@ check_balance(ErtsRunQueue *c_rq)
     ErtsMigrationPaths *new_mpaths, *old_mpaths;
     ErtsRunQueueBalance avg = {0};
     Sint64 scheds_reds, full_scheds_reds;
-    int forced, active, current_active, oowc, half_full_scheds, full_scheds,
+    int forced, active, current_active, half_full_scheds, full_scheds,
 	mmax_len, blnc_no_rqs, qix, pix, freds_hist_ix;
 #if ERTS_HAVE_SCHED_UTIL_BALANCING_SUPPORT
     int sched_util_balancing;
@@ -4877,7 +4876,6 @@ check_balance(ErtsRunQueue *c_rq)
 	run_queue_info[qix].full_reds_history_change
 	    = rq->full_reds_history[freds_hist_ix];
 
-	run_queue_info[qix].oowc = rq->out_of_work_count;
 	run_queue_info[qix].max_len = rq->max_len;
 	rq->check_balance_reds = INT_MAX;
 
@@ -4893,7 +4891,6 @@ check_balance(ErtsRunQueue *c_rq)
     half_full_scheds = 0;
     full_scheds_reds = 0;
     scheds_reds = 0;
-    oowc = 0;
     mmax_len = 0;
 
     /* Calculate availability for each priority in each run queues */
@@ -4950,7 +4947,6 @@ check_balance(ErtsRunQueue *c_rq)
 	}
 	run_queue_info[qix].reds = treds;
 	scheds_reds += treds;
-	oowc += run_queue_info[qix].oowc;
 	if (mmax_len < run_queue_info[qix].max_len)
 	    mmax_len = run_queue_info[qix].max_len;
     }

--- a/erts/emulator/beam/erl_process.c
+++ b/erts/emulator/beam/erl_process.c
@@ -4654,7 +4654,8 @@ do {									\
     run_queue_info[(QIX)].full_reds_history_change = (LAST_REDS);	\
 } while (0)
 
-#define ERTS_DBG_CHK_FULL_REDS_HISTORY(RQ)				\
+#ifdef DEBUG
+#  define ERTS_DBG_CHK_FULL_REDS_HISTORY(RQ)				\
 do {									\
     int sum__ = 0;							\
     int rix__;								\
@@ -4662,6 +4663,9 @@ do {									\
 	sum__ += (RQ)->full_reds_history[rix__];			\
     ASSERT(sum__ == (RQ)->full_reds_history_sum);			\
 } while (0);
+#else
+#  define ERTS_DBG_CHK_FULL_REDS_HISTORY(RQ)
+#endif
 
 #define ERTS_PRE_ALLOCED_MPATHS 8
 

--- a/erts/emulator/sys/common/erl_mmap.c
+++ b/erts/emulator/sys/common/erl_mmap.c
@@ -1897,7 +1897,7 @@ erts_mremap(ErtsMemMapper* mm,
 #endif
 #ifdef ERTS_HAVE_OS_MREMAP
 	if (superaligned) {
-	    return remap_move(mm, flags, new_ptr, old_size, sizep);
+	    return remap_move(mm, flags, ptr, old_size, sizep);
 	} else {
 	    new_ptr = os_mremap(ptr, old_size, asize, 0);
 	    if (!new_ptr)

